openedx/frontend-app-course-authoring
# fix: datatable state persistence issues

This PR fixes two issues:
1. Many states do not persist after performing actions that causes a change to the files in the table (changing the lock, checking the info and delete). The states affected include filters and page index.
2. Search filter isn't displaying correctly on the filter status bar. Example below shows the issue with 'banner' shown as 'b'.
<img width="258" alt="image" src="https://github.com/openedx/frontend-app-course-authoring/assets/56318341/f7ab34c6-98d1-41b1-af13-b9c7ac134254">



Testing instructions for filter persisting:
1. Navigate to the files page https://course-authoring.stage.edx.org/course/{courseId}/assets
2. Enter a search string.
3. Open the dropdown options for a file and click Lock (or Unlock). Filters should persist.
4. Open the dropdown options for a file and click Info. Close the modal. Filters should persist.
5. Open the dropdown options for a file and click delete. Confirm deletion. Filters should persist.
6. Open the Sort and Filter menu. Enter any number of filters.
7. Repeat steps 3-5. Filters you entered should persist.
8. Navigate to the videos page https://course-authoring.stage.edx.org/course/{courseId}/videos
9. Enter a combination of search string and filters options like above.
10. Repeat steps 4-5. Filters you entered should persist.

Testing instructions for page index persisting:
1. Navigate to the files page https://course-authoring.stage.edx.org/course/{courseId}/assets
2. Go to the 2nd page of the DataTable (must have over 50 files).
3. Open the dropdown options for a file and click Lock (or Unlock). Table should remain on 2nd page.
4. Open the dropdown options for a file and click Info. Close the modal. Table should remain on 2nd page.
5. Open the dropdown options for a file and click delete. Confirm deletion. Table should remain on 2nd page.
6. Repeat for videos page https://course-authoring.stage.edx.org/course/{courseId}/videos

Testing instructions for search filter displaying correctly:
1. Navigate to the files page https://course-authoring.stage.edx.org/course/{courseId}/assets
4. Enter a search string longer than 1 character.
5. The string you entered should display below.
7. Repeat with videos page https://course-authoring.stage.edx.org/course/{courseId}/videos


2U ticket: https://2u-internal.atlassian.net/browse/TNL-11259
